WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -262,6 +262,32 @@ extension OrderDetailsViewModel {
             self.syncSubscriptions { _ in
                 group.leave()
             }
+
+            // Shipping labels need to be synced after the order but before we complete
+            // the order sync group to ensure the UI shows the latest data
+            group.enter()
+            Task { @MainActor [weak self] in
+                defer {
+                    group.leave()
+                }
+                // Check Woo Shipping support first, to ensure correct flows are enabled for shipping labels.
+                self?.dataSource.isEligibleForWooShipping = ((await self?.isWooShippingSupported()) != nil)
+
+                // Check creation eligibility
+                let isEligible = await self?.checkShippingLabelCreationEligibility()
+                self?.dataSource.isEligibleForShippingLabelCreation = isEligible ?? false
+
+                // Sync shipping labels and update order with the result if available
+                if let shippingLabels = await self?.syncShippingLabels() {
```

Comment:
Should we call `syncShippingLabels` even if `checkShippingLabelCreationEligibility` returned false above? If yes, can we start this request asynchronously without waiting for the response from `checkShippingLabelCreationEligibility`?